src/loader.rs

```diff
@@ -16,6 +16,34 @@ impl Loader {
       path: path.to_owned(),
       io_error,
     })?;
-    Ok(self.arena.alloc(src))
+
+    // Parse include directives.
+    // Include directives are lines that look like:
+    // #include "relative/path/to/file"
+    let include_regexp = Regex::new(r#"^#include "([^"]+)"$"#).unwrap();
```

Comment:
Let's extract this logic to a separate, testable function which we call from here.